### PR TITLE
Bindings commercetools

### PR DESCRIPTION
**Pull request by harrykimpel**
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [x] [Read the contribution guide](https://docs.dapr.io/contributing/contributing-docs/)
- [x] Commands include options for Linux, MacOS, and Windows within codetabs
- [x] New file and folder names are globally unique
- [x] Page references use shortcodes instead of markdown or URL links
- [x] Images use HTML style and have alternative text
- [x] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

Adding documentation for the commercetools output binding.

## Issue reference

#2351

### CODE REVIEW

**Comment by harrykimpel** [comment]
the validate-links checker fails for the GitHub Codespaces overview page with a 403

File: daprdocs/content/en/reference/components-reference/supported-bindings/commercetools.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: <NAME>
+  namespace: <NAMESPACE>
+spec:
+  type: bindings.commercetools
+  version: v1
+  metadata:
+  - name: Region # required.
```
Comment (berndverst) [inline daprdocs/content/en/reference/components-reference/supported-bindings/commercetools.md]: Comparing to your code PR all these meatadata fields must start with a lowercase character

**Review by berndverst — CHANGES_REQUESTED** [review]

File: daprdocs/content/en/reference/components-reference/supported-bindings/commercetools.md (hunk 1 above)
Comment (yaron2) [reply daprdocs/content/en/reference/components-reference/supported-bindings/commercetools.md]: ping @harrykimpel

Comment (harrykimpel) [reply daprdocs/content/en/reference/components-reference/supported-bindings/commercetools.md]: thanks @yaron2, will update

Comment (harrykimpel) [reply daprdocs/content/en/reference/components-reference/supported-bindings/commercetools.md]: done

**Comment by greenie-msft** [comment]
Hi @harrykimpel,

It looks like some additional files were checked-in unintentionally. Can you please revert those files (dccache ?)

Also, I noticed you added a link to a sample in your GitHub account. Are you open to submitting that to the official Samples repo in the Dapr org?

**Comment by harrykimpel** [comment]
> Hi @harrykimpel,
> 
> It looks like some additional files were checked-in unintentionally. Can you please revert those files (dccache ?)
> 
> Also, I noticed you added a link to a sample in your GitHub account. Are you open to submitting that to the official Samples repo in the Dapr org?

@greenie-msft I removed the dccache file. I am also happy to submit the sample to the official sample repo. I wasn't sure whether this would belong there. are you referring to this repo https://github.com/dapr/samples or the quickstarts?

**Comment by msfussell** [comment]
>> I wasn't sure whether this would belong there. are you referring to this repo https://github.com/dapr/samples or the quickstarts?

Yes, submit to this Samples repo. The expectation on this repo is that the submitted chooses to keep the sample current and maintained. If these go very stale over time (i.e. Feature removed or does not compile) they get removed. 

Please update this link in the doc- [Sample app](https://github.com/harrykimpel/dapr-commercetools-sample) that leverages the commercetools binding with sample GraphQL query and then resubmit.

**Comment by harrykimpel** [comment]
> > > I wasn't sure whether this would belong there. are you referring to this repo https://github.com/dapr/samples or the quickstarts?
> 
> Yes, submit to this Samples repo. The expectation on this repo is that the submitted chooses to keep the sample current and maintained. If these go very stale over time (i.e. Feature removed or does not compile) they get removed.
> 
> Please update this link in the doc- [Sample app](https://github.com/harrykimpel/dapr-commercetools-sample) that leverages the commercetools binding with sample GraphQL query and then resubmit.

@msfussell I added the samples app to the samples repo and updated the link in the docs

**Comment by harrykimpel** [comment]
Tried to update the link to the Azure SQL DB docs, but now I am getting a fail for a localized version

**Comment by msfussell** [comment]
@harrykimpel - See comment on the Sample repo PR here https://github.com/dapr/samples/pull/106

File: daprdocs/content/en/reference/components-reference/supported-bindings/commercetools.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+```
+{{% alert title="Warning" color="warning" %}}
+The above example uses secrets as plain strings. It is recommended to use a secret store for the secrets as described [here]({{< ref component-secrets.md >}}).
+{{% /alert %}}
+
+## Spec metadata fields
+
+| Field              | Required | Binding support |  Details | Example |
+|--------------------|:--------:|------------|-----|---------|
+| Region | Y | Output | The region of the commercetools project | `"europe-west1"` |
```
Comment (berndverst) [inline daprdocs/content/en/reference/components-reference/supported-bindings/commercetools.md]: These fields should all start with a lowercase character to match the code PR:

https://github.com/dapr/components-contrib/pull/1652/files

**Review by berndverst — CHANGES_REQUESTED** [review]

**Comment by harrykimpel** [comment]
> @harrykimpel - See comment on the Sample repo PR here [dapr/samples#106](https://github.com/dapr/samples/pull/106)

@msfussell done

File: daprdocs/content/en/reference/components-reference/supported-bindings/commercetools.md (hunk 2 above)
Comment (harrykimpel) [reply daprdocs/content/en/reference/components-reference/supported-bindings/commercetools.md]: @berndverst done

**Review by msfussell — APPROVED** [review]
LGTM. Thanks for the contribution!

File: daprdocs/content/en/reference/components-reference/supported-bindings/commercetools.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+
+## Related links
+
+- [Basic schema for a Dapr component]({{< ref component-schema >}})
+- [Bindings building block]({{< ref bindings >}})
+- [How-To: Trigger application with input binding]({{< ref howto-triggers.md >}})
+- [How-To: Use bindings to interface with external resources]({{< ref howto-bindings.md >}})
+- [Bindings API reference]({{< ref bindings_api.md >}})
+- [Sample app](https://github.com/dapr/samples/commercetools-graphql-sample) that leverages the commercetools binding with sample GraphQL query
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-bindings/commercetools.md]: Given I have merged the Sample into the samples repo, the URL for this is https://github.com/dapr/samples/tree/master/commercetools-graphql-sample

Can you confirm this is your expectation, and then will have to change this

```suggestion
- [Sample app](https://github.com/dapr/samples/tree/master/commercetools-graphql-sample) that leverages the commercetools binding with sample GraphQL query
```

**Comment by msfussell** [comment]
@harrykimpel - Just found the link issue above. PTAL?

**Comment by msfussell** [comment]
Thanks for the amazing contribution!!

**Pull request merged**